src/relay/backend/build_module.cc

```diff
@@ -0,0 +1,669 @@
+      });
+    } else if (name == "set_fallback_device") {
+      return PackedFunc([sptr_to_self, this](TVMArgs args, TVMRetValue* rv) {
+        std::string dev = args[0];
+        this->_SetFallBackDev(dev);
+      });
+    } else if (name == "add_pass") {
+      return PackedFunc([sptr_to_self, this](TVMArgs args, TVMRetValue* rv) {
+        std::string pass_name = args[0];
+        this->_AddPass(pass_name);
+      });
+    } else if (name == "disable_pass") {
+      return PackedFunc([sptr_to_self, this](TVMArgs args, TVMRetValue* rv) {
+        std::string pass_name = args[0];
+        this->_DisablePass(pass_name);
+      });
+    } else {
+      return PackedFunc([sptr_to_self](TVMArgs args, TVMRetValue* rv) {});
+    }
+  }
+
+  /*!
+   * \brief Get the GraphJSON for runtime
+   *
+   * \return const std::string graph_json
+   */
+  const std::string _GetGraphJSON() {
+    return ret_.graph_json;
+  }
+  /*!
+   * \brief Add extra pass during build
+   * 
+   * \param pass_name 
+   */
+  void _AddPass(const std::string& pass_name) {
+    cfg_.add_pass.insert(pass_name);
+  }
+
+  void _DisablePass(const std::string& pass_name) {
+    cfg_.disabled_pass.insert(pass_name);
+  }
+
+  void _SetFallBackDev(const std::string& dev) {
+    cfg_.fall_back_device = dev;
+  }
+  /*!
+   * \brief Get the Module object
+   *
+   * \return runtime::Module
+   */
+  runtime::Module _GetModule() {
+    return ret_.mod;
+  }
+
+  /*!
+   * \brief List all paramter names
+   * 
+   * \return Array<StringImm> 
+   */
+  Array<HalideIR::Expr> _ListParamNames() {
+    Array<HalideIR::Expr> ret;
+    for (const auto& kv : params_) {
+      ret.push_back(ir::StringImm::make(kv.first));
+    }
+    return ret;
+  }
+
+  /*!
+   * \brief Get the Param of name
+   * 
+   * \param name 
+   * \return runtime::NDArray 
+   */
+  runtime::NDArray _GetParam(const std::string& name) {
+    CHECK_GT(params_.count(name), 0) << "Can not find param with name: " << name;
+    return params_[name];
+  }
+
+  /*!
+   * \brief Set the parameters
+   *
+   * \param name name of parameter
+   * \param data_in input DLTensor
+   */
+  void _SetParams(const std::string& name, DLTensor* data_in) {
+    if (!params_.count(name)) {
+      std::vector<int64_t> shape(data_in->shape, data_in->shape + data_in->ndim);
+      params_[name] = tvm::runtime::NDArray::Empty(shape, data_in->dtype, {kDLCPU, 0});
+    }
+    params_[name].CopyFrom(data_in);
+  }
+
+  /*!
+   * \brief Set the optimization level
+   *
+   * \param level
+   */
+  void _SetOptLevel(char level) {
+    cfg_.opt_level = level;
+  }
+
+  /*!
+   * \brief type key
+   *
+   * \return const char*
+   */
+  const char* type_key() const final {
+    return "RelayBuildModule";
+  }
+
+  /*!
+   * \brief Build relay function for graph runtime
+   *
+   * \param func Relay Function
+   * \param target Target device
+   * \param target_host Host target device
+   */
+  void _Build(Function func,
```

Comment:
@tqchen given C++ style I think we should remove the leading underscore and instead just mark the APIs as private, thoughts?